oak_channel/SPEC.md

```diff
@@ -138,149 +93,80 @@ A byte encoded frame MUST consist of the following fields:
  0                   1                   2                   3
  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                            padding                            |
+|                         invocation_id                         |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
```

Comment:
Maybe a nit, but I think it would be nice to keep length as the first field for consistency and to simplify the low level parsing logic, and also potentially allow stacking combinators to express this transformation (e.g. receive byte stream, cut it into frames by just looking at the length field and send the content to the next combinator, etc.). We don't have to actually implement it that way, but I think it's nice if it could be done from the outside in if desired.

Reply:
I think we will always parse the entire header for a layer before parsing the bytes, so the exact order doesn't matter. But I have swapped the body length and the invocation ID so the two length fields are adjacent.